.travis.yml

```diff
@@ -16,33 +18,36 @@ env:
 services:
   - mongodb
 
-before_script:
-  - sleep 15 # mongo takes time to start
-  - mongo mydb_test --eval 'db.createUser({user:"travis",pwd:"test",roles:["readWrite"]});'
+jobs:
+  include:
+    - name: "Run Linter (black)"
+      python: 3.9
+      install: pip install --upgrade black
+      script: black .
+    - name: "Run tests on python 3.7"
+      python: 3.7
+    - name: "Run tests on python 3.8"
+      python: 3.8
+    - name: "Run tests on python 3.9"
+      python: 3.9
+      env: NUMPY_VERSION=1.19.2
+  allow_failures:
+    - name: "Run tests on python 3.9"
 
 install:
-  - sudo apt-get remove ipython
+  - sudo apt-get remove ipython || true  # ipython is not pre-installed in Focal
   - pip install --upgrade pip setuptools wheel
   - pip install $IPYTHON
   - pip install --only-binary=numpy,scipy numpy==$NUMPY_VERSION scipy
   - pip install -e '.[MongoTrials, SparkTrials, ATPE, dev]'
   - pip install nose
   - pip install pyspark==$SPARK_VERSION
 
-jobs:
-  include:
-    - stage: "lint"
-      name: "Run Linter (black)"
-      python: 3.8
-      script: black .
-    - stage: "test python 3.7"
-      name: "Run tests on python 3.7"
-      script: pytest
-      python: 3.7
-    - stage: "test python 3.8"
-      name: "Run tests on python 3.8"
-      script: pytest
-      python: 3.8
+before_script:
+  - sleep 15 # mongo takes time to start
```

Comment:
In my experience, it often takes more than 15 seconds to install mongodb. It seems better to increase this time to 30seconds.

Reply:
mongodb should be kicked off on line 19 and then we pip install a lot of stuff and that takes more than a full minute...

Look at the times along the right margin of the Travis output...
<img width="914" alt="Screenshot 2020-10-27 at 14 51 20" src="https://user-images.githubusercontent.com/3709715/97311154-595d1200-1864-11eb-88eb-b5ec74fea1a5.png">